### PR TITLE
update and remove cart

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
[- ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/73598)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controllers/Cart.controller.ts [hunk 1]
```diff
@@ -27,68 +25,124 @@ const getUserCart = async (userId: number, res: Response) => {
...
 
-  const { total, items } = rawResult;
+export const deleteCartItem = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    await cartItemRepository.delete(parseInt(req.params.id));
+    res.redirect('/cart/list');
```
Comment (huongnt-2545) [inline src/controllers/Cart.controller.ts:142]: để ng dùng có trải nghiệm tốt khi thao tác thì phần remove item from cart này em dùng ajax để tránh load page nha

File: src/public/js/cart.js [hunk 2]
```diff
@@ -1,78 +1,124 @@
...
+      }
     });
   });
+});
+
+// Lấy tham chiếu đến nút cập nhật giỏ hàng
+const updateCartBtn = document.getElementById('update-cart-btn');
+
+// Thêm sự kiện click vào nút cập nhật giỏ hàng
+updateCartBtn.addEventListener('click', function (event) {
```
Comment (huongnt-2545) [inline src/public/js/cart.js:49]: sao ko dùng jquery cho code ngắn gọn?

File: src/controllers/Cart.controller.ts (hunk 1 above)
Comment (ghost) [reply src/controllers/Cart.controller.ts:142]: Vâng ạ!

File: src/public/js/cart.js (hunk 2 above)
Comment (ghost) [reply src/public/js/cart.js:49]: Vâng em sẽ sửa lại ạ!

**Comment by ghost** [comment]
ready

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**